exonum/src/runtime/migrations.rs

```diff
@@ -76,11 +81,19 @@
 //! returns to the [`Stopped`] status. The local migration result is ignored; if the migration
 //! script has not completed locally, it is aborted.
 //!
+//! Note that commitment and flushing are separate operations and need to be performed in
+//! different blocks. When a migration is flushed, the migrated data needs to have a definite
+//! state, which is ensured by an earlier commitment acting as a [Great Filter]. The requirement
```

Comment:
>acting as a Great Filter

Philosophy is good, but I think this analogy makes an already complex topic even harder to get :)